File: toolloop-front/src/app/core/models/dto/add-tool-request.ts
import { ToolAvailabilityDTO } from "./tool-availability-dto";

export interface AddToolRequest {
    name: string;
    description: string;
    pricePerDay: number;
    securityDeposit: number;
    categoryId: number;
    condition: string;
    photoKeys: string[];
    availability: ToolAvailabilityDTO;
}

File: toolloop-back/src/main/java/com/toolloop/model/dto/AddToolRequest.java
package com.toolloop.model.dto;

import io.quarkus.runtime.annotations.RegisterForReflection;

import java.math.BigDecimal;
import java.util.List;

@RegisterForReflection
public record AddToolRequest(
    String name,
    String description,
    BigDecimal pricePerDay,
    BigDecimal securityDeposit,
    Long categoryId,
    String condition,
    List<String> photoKeys,
    ToolAvailabilityDTO availability
) { }
